pkg/controller/sdk/finalizers_accessor.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sdk
+
+import (
+	"reflect"
+)
+
+// FinalizersAccessor is the interface for a Resource that implements the getter and setting for
+// accessing its Finalizer set.
+// +k8s:deepcopy-gen=true
+type FinalizersAccessor interface {
+	GetFinalizers() interface{}
+	SetFinalizers(finalizers interface{})
+}
+
+// NewReflectedFinalizersAccessor uses reflection to return a FinalizersAccessor to access the field
+// called "Finalizers".
+func NewReflectedFinalizersAccessor(object interface{}) FinalizersAccessor {
+	objectValue := reflect.Indirect(reflect.ValueOf(object))
+
+	// If object is not a struct, don't even try to use it.
+	if objectValue.Kind() != reflect.Struct {
+		return nil
+	}
+
+	finalizerField := objectValue.FieldByName("Finalizers")
```

Comment:
Should this dig through ObjectMeta as well, or no?

Reply:
No. ObjectMeta is added to the struct via [embedding](https://github.com/knative/eventing-sources/blob/58dd271c37f4b085ccaf39ca61e6975f699330bc/pkg/apis/sources/v1alpha1/containersource_types.go#L131), so there is no ObjectMeta to dig through.